### PR TITLE
Added a new ``list_data_files`` function that can be used to search for files recursively

### PR DESCRIPTION
**Pull request by astrofrog**
In the package-template, we currently search for c files manually:

```
c_files = []
for root, dirs, files in os.walk(PACKAGENAME):
    for filename in files:
        if filename.endswith('.c'):
            c_files.append(
                os.path.join(
                    os.path.relpath(root, PACKAGENAME), filename))
```

we now also need to search for all files in ``packagename/data/`` recursively, so I thought it would be more convenient to write a function to do this.

I can add tests if we think this is a good idea.

cc @embray @mdboom

### CODE REVIEW

**Comment by astrofrog** [comment]
@embray - I've now updated this to use the Python 3.5 recursive glob syntax (in fact, on Python 3.5 it will use the real glob). Does this seem like a good plan? If you are ok with this, I can add tests.

File: astropy_helpers/utils.py [hunk 1]
```diff
@@ -660,3 +662,35 @@ def minversion(module, version, inclusive=True, version_path='__version__'):
...
+
+
+def find_data_files(package, pattern):
+    """
+    Include files matching ``pattern`` inside ``package``.
+
+    Parameters
+    ----------
+    package : str
+        The pacakge inside which to look for data files
```
Comment (embray) [inline astropy_helpers/utils.py]: "pacakge"

**Comment by embray** [comment]
Yeah, I think this fine.

File: astropy_helpers/utils.py (hunk 1 above)
Comment (astrofrog) [inline astropy_helpers/utils.py]: Fixed

**Comment by astrofrog** [comment]
I added a unit test - this is now ready for review.

File: astropy_helpers/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            f.write('test')
+
+    filenames = find_data_files(data.strpath, '**/*.dat')
+
+    filenames = sorted(os.path.relpath(x, data.strpath) for x in filenames)
+
+    assert filenames[0] == 'data.dat'
+    assert filenames[1] == 'sub1/data.dat'
+    assert filenames[2] == 'sub1/sub3/data.dat'
+    assert filenames[3] == 'sub2/data.dat'
```
Comment (embray) [inline astropy_helpers/tests/test_utils.py]: should probably add something like `filenames = [f.replace(os.sep, '/') for f in filenames]`

Comment (astrofrog) [inline astropy_helpers/tests/test_utils.py]: Fixed - I just used ``os.path.join``

**Comment by embray** [comment]
Needs to be rebased for some reason but then I'll merge.

**Comment by astrofrog** [comment]
@embray - done. It was a conflict with ``classproperty`` which was added in the mean time.

**Comment by embray** [comment]
Changelog entry?

**Comment by embray** [comment]
I'll just add one manually, nbd.

**Pull request merged**